PipeMethodCalls/Endpoints/PipeServer.cs

```diff
@@ -33,6 +33,17 @@ public PipeServer(string pipeName, Func<THandling> handlerFactoryFunc, PipeOptio
 			this.options = options;
 		}
 		
+		/// <summary>
+		/// Initializes a new instance of the <see cref="PipeServer{THandling}"/> class.
+		/// </summary>
+		/// <param name="existingPipe">Instance of a named pipe to upgrade.</param>
+		/// <param name="handlerFactoryFunc">A factory function to provide the handler implementation.</param>
+		public PipeServer(NamedPipeServerStream existingPipe, Func<THandling> handlerFactoryFunc)
+		{
+			this.rawPipeStream = existingPipe;

```

Comment:
There are a couple settings that need a particular value in order for the method calls to work.

- We need the PipeOptions.Asynchronous flag to be set
- We need PipeDirection.InOut

Could you throw an ArgumentException if we've been handed a pipe stream with the wrong settings?

Reply:
Yeah, I will look at doing that.  I agree that is a good safety measure to have in place.

Reply:
Oh and TransmissionMode needs to be PipeTransmissionMode.Byte. It won't work with PipeTransmissionMode.Message.

Reply:
Also can you mention these restrictions in the docs?

Reply:
Last commit should cover these cases.  I noticed the `PipeTransmissionMode.Byte` restriction and threw it in already 😄.  I added a small snippet to the exception section of the docs.  Let me know what you think.  Happy to revisit wording/implementation.  I am unable to get the actual enum values (other than Transmission mode), but `CanWrite`, `CanRead`, and `IsAsync` should be what we want (best I can tell from the docs and testing).